### PR TITLE
SSH Key: Support additional encryption algorithms

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM1**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: website/docs/r/is_ssh_key.html.markdown [hunk 1]
```diff
@@ -40,6 +40,7 @@ Review the argument references that you can specify for your resource.
   **&#x2022;** For more information, about creating access tags, see [working with tags](https://cloud.ibm.com/docs/account?topic=account-tag&interface=ui#create-access-console).</br>
   **&#x2022;** You must have the access listed in the [Granting users access to tag resources](https://cloud.ibm.com/docs/account?topic=account-access) for `access_tags`</br>
   **&#x2022;** `access_tags` must be in the format `key:value`.
+- `type` - (Optional, String) The crypto system used by this key. </br> Allowed values are : [`ed25519`, `rsa`].
```
Comment (deepaksibm) [inline website/docs/r/is_ssh_key.html.markdown:43]: mention that default is 'rsa'

Comment (deepaksibm) [reply website/docs/r/is_ssh_key.html.markdown:43]: also update the example to include 'type'

**Comment by deepaksibm** [comment]
update doc at website/docs/d/is_ssh_keys.html.markdown
It says allowed values are rsa. Either remove that constraint line or mention both types

**Comment by deepaksibm** [comment]
looks good.

**Comment by SunithaGudisagarIBM1** [comment]
resource "ibm_is_ssh_key" "isExampleKey" {
  name       = "sunitha-tfp-test"
  public_key = <<-EOT
            ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHEE9sLlndKFR/hVbF7SUNhKBFrxscJDHrVN/OD1Z+8V abc.edf@ibm.com
        EOT
  type       = "ed25519"
} 



/* resource "ibm_is_ssh_key" "isExampleKeyWithSignature" {
  name       = "sunitha-rsa-ssh-key-tfp"
  public_key = <<-EOT
            ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCKVmnMOlHKcZK8tpt3MP1lqOLAcqcJzhsvJcjscgVERRN7/9484SOBJ3HSKxxNG5JN8owAjy5f9yYwcUg+JaUVuytn5Pv3aeYROHGGg+5G346xaq3DAwX6Y5ykr2fvjObgncQBnuU5KHWCECO/4h8uWuwh/kfniXPVjFToc+gnkqA+3RKpAecZhFXwfalQ9mMuYGFxn+fwn8cYEApsJbsEmb0iJwPiZ5hjFC8wREuiTlhPHDgkBLOiycd20op2nXzDbHfCHInquEe/gYxEitALONxm0swBOwJZwlTDOB7C6y2dzlrtxr1L59m7pCkWI4EtTRLvleehBoj3u7jB4usR  abc@abc-MacBook-Pro.local
        EOT
} */

**Comment by SunithaGudisagarIBM1** [comment]
```
resource "ibm_is_ssh_key" "isExampleKey" {
  name = "sunitha-tfp-test"
  /* public_key = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHEE9sLlndKFR/hVaF7SUNhKBFrxscJDHrVN/OD1Z+8V sunitha.gudisagar@ibm.com" */
  public_key = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIBSht2fMPw3MnUSfETUw55BnFAxZ+YV2KyX1m3qcTtfv sunitha.gudisagar.com"
  type       = "ed25519"
}

resource "ibm_is_ssh_key" "isExampleKeyWithSignature" {
  name       = "sunitha-rsa-ssh-key-tfp-1-checking"
  public_key = <<-EOT
            ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCkF5mR1NqO7z6wviLUY2JZAWcEBhOmYi8lNlrLKP1mI4XOLUFq5KeIy5Ta14wqQR/W2dA72E51ummWn+39z9vgWgYKY5yIuVYy6b2W5Mo2MM73iEi1mB+GgqXWRTGRFBAPgT5mnqiwPZE0yKvQAe5vLPx6z4BxWKNi5Tp31Hf0KKHLKzrH4zUGSyoEFFYeoxUO8YSc6oyAKhJQBQAFXl0u9YSnuyiXUqYDpa06w0PUl2M6ALBTVa1uLmacTIJxJhJzDJxidW3fIVO9Nm7RSU0WcS2FGyqP3jNQKLAhpqUyDOyWfV/2yjy/H4Y3gjfLKCIhZIjUVO1KJlWcSScoJo7ILPWnzmr0uWmC/4qrqU0I3+8/AdBvJF1xCpzDBb/nPKWT2m8s5ZyahpmP4F91K6J0Zk3KLWzXkFrrB4MBpW9LLitPmgFhBitGKmD+Y2w93F3SvdzeLG4MlXg6RBSEJCFdWC9q+xPHxHg9TDkDVr6GoxOS/a/x8MLTHqnBhgczvxk= sunitha@Sunithas-MacBook-Pro.local
        EOT
}


resource "ibm_is_vpc" "testacc_vpc" {
  name = "sunitha-vpc"
}

resource "ibm_is_subnet" "testacc_subnet" {
  name            = "sunitha-subnet"
  vpc             = ibm_is_vpc.testacc_vpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
}

/* data "ibm_is_security_group_rules" "example" {
	security_group = ibm_is_vpc.testacc_vpc.default_security_group
} */

resource "ibm_is_security_group_rule" "testacc_security_group_rule_all" {
  group     = ibm_is_vpc.testacc_vpc.default_security_group
  direction = "inbound"
  /* remote    = "127.0.0.1" */
}
data "ibm_is_security_group_rules" "example" {
  depends_on = [
    ibm_is_security_group_rule.testacc_security_group_rule_all
  ]
	security_group = ibm_is_vpc.testacc_vpc.default_security_group
}

/* data "ibm_is_images" "test1" {
  status = "available"
} */

resource "ibm_is_instance" "testacc_instance" {
  name    = "sunitha-instance"
  image   = "r134-e4156ce6-3d9d-4d3f-a086-0a1440131157"
  profile = "bx2-2x8"
  primary_network_interface {
    subnet = ibm_is_subnet.testacc_subnet.id
  }
  vpc  = ibm_is_vpc.testacc_vpc.id
  zone = "us-south-1"
  keys = [ibm_is_ssh_key.isExampleKey.id]
  network_interfaces {
    subnet = ibm_is_subnet.testacc_subnet.id
    name   = "eth1"
  }
}

resource "ibm_is_floating_ip" "testacc_floatingip" {
  name   = "floating-ip-sunitha"
  target = ibm_is_instance.testacc_instance.primary_network_interface[0].id
}

```
     id   = "r134-5c33dc64-8e0a-491f-8c97-5de33612b6c2"
     name  = "ibm-redhat-8-3-minimal-amd64-4" 

      id  = "r134-e4156ce6-3d9d-4d3f-a086-0a1440131157"
      name  = "ibm-centos-stream-9-amd64-2" 

Screenshot 2023-04-26 at 7 54 40 PM

Screenshot 2023-04-27 at 8 44 43 PM

**Comment by SunithaGudisagarIBM1** [comment]
Windows testing:

resource "ibm_is_instance" "testacc_instance" {
  name    = "sunitha-instance"
  image   = "r134-bb0f1d19-afcd-4730-af52-f7eea6126549"
  profile = "bx2-2x8"
  primary_network_interface {
    subnet = ibm_is_subnet.testacc_subnet.id
  }
  vpc  = ibm_is_vpc.testacc_vpc.id
  zone = "us-south-1"
  keys = [ibm_is_ssh_key.isExampleKey.id]
  network_interfaces {
    subnet = ibm_is_subnet.testacc_subnet.id
    name   = "eth1"
  }
}

As ed25519 is not supported for windows we get the below error:
Screenshot 2023-05-02 at 10 55 37 AM

File: ibm/service/vpc/resource_ibm_is_ssh_key.go [hunk 2]
```diff
@@ -466,11 +474,11 @@ func parseKey(s string) (ssh.PublicKey, error) {
...
-	// - ssh-rsa <base64 key> <comment>
-	// if PublicKey provides other than just base64 key, then first part must be "ssh-rsa"
-	if subStrs := strings.Split(s, " "); len(subStrs) > 1 && subStrs[0] != "ssh-rsa" {
-		return nil, errors.New("not an RSA key")
+	// - ssh-rsa/ssh-ed25519 <base64 key>
+	// - ssh-rsa/ssh-ed25519 <base64 key> <comment>
+	// if PublicKey provides other than just base64 key, then first part must be "ssh-rsa" or "ssh-ed25519"
+	if subStrs := strings.Split(s, " "); len(subStrs) > 1 && subStrs[0] != "ssh-rsa" && subStrs[0] != "ssh-ed25519" {
+		return nil, errors.New("not an RSA key OR ED25519 key")
 	}
```
Comment (sumitkumartiwari) [inline ibm/service/vpc/resource_ibm_is_ssh_key.go:482]: This validation should be depend on the keyType.

**Comment by ujjwal-ibm** [comment]
migrating to upstream

**Pull request closed without merging**